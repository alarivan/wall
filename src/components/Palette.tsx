import React from 'react';
import styled from 'styled-components';
import PaletteCell from './PaletteCell';
import { TColor } from '../types';

interface Props {
  colors: TColor[];
  current: TColor;
  onColorClick: (color: TColor) => void;
}
const StyledPalette = styled.div`
  display: grid;
  grid-gap: 2px;
  grid-template-columns: repeat(3, 20px);
  grid-auto-rows: 20px;
  align-self: start;
  background: #c2c2d1;
  padding: 0.25rem;
`;

const Palette: React.FC<Props> = ({ onColorClick, current, colors }) => {
  return (
    <StyledPalette>
      {colors.map((color, index) => (
        <PaletteCell
          data-testid='palette-cell'
          key={index}
          onClick={() => onColorClick(color)}
          cellColor={color}
          active={current === color}
        />
      ))}
    </StyledPalette>
  );
};

export default Palette;
