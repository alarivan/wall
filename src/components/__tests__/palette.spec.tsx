import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Palette from '../Palette';

afterEach(cleanup);

describe('Palette', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Palette
        colors={['white', 'blue', 'red']}
        current={'red'}
        onColorClick={jest.fn()}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onColorClick', () => {
    const onColorClick = jest.fn();
    const { queryAllByTestId } = render(
      <Palette
        colors={['white', 'blue', 'red']}
        current={'red'}
        onColorClick={onColorClick}
      />,
    );

    fireEvent.click(queryAllByTestId('palette-cell')[0]);

    expect(onColorClick).toHaveBeenCalledTimes(1);
  });
});
