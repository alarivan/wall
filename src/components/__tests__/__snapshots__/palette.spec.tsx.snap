// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  background: white;
  border-style: solid;
  border-color: white;
  border-width: 0 1px 1px 0;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: none;
  -webkit-transition: border-radius 50ms ease-in-out;
  transition: border-radius 50ms ease-in-out;
  height: 100%;
}

.c1:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c1:hover {
  border-width: 0;
  border-radius: none;
}

.c2 {
  background: blue;
  border-style: solid;
  border-color: blue;
  border-width: 0 1px 1px 0;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: none;
  -webkit-transition: border-radius 50ms ease-in-out;
  transition: border-radius 50ms ease-in-out;
  height: 100%;
}

.c2:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c2:hover {
  border-width: 0;
  border-radius: none;
}

.c3 {
  background: red;
  border-style: solid;
  border-color: red;
  border-width: 0 1px 1px 0;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  -webkit-transition: transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  cursor: pointer;
  box-shadow: 1px 1px 5px 0px rgba(0,0,0,0.75);
  border-style: solid;
  border-color: black;
  border-width: 0;
  border-radius: 50%;
  -webkit-transition: border-radius 50ms ease-in-out;
  transition: border-radius 50ms ease-in-out;
  height: 100%;
}

.c3:hover {
  border-width: 1px;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c3:hover {
  border-width: 0;
  border-radius: 50%;
}

.c0 {
  display: grid;
  grid-gap: 2px;
  grid-template-columns: repeat(3,20px);
  grid-auto-rows: 20px;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  background: #c2c2d1;
  padding: 0.25rem;
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="palette-cell"
    />
    <div
      class="c2"
      data-testid="palette-cell"
    />
    <div
      class="c3"
      data-testid="palette-cell"
    />
  </div>
</DocumentFragment>
`;
